tests/application-operator-tests/test/applicationcontroller/suite.go

```diff
@@ -64,6 +72,59 @@ func (ts *TestSuite) CreateApplication(t *testing.T, accessLabel string, skipIns
 	require.NotNil(t, application)
 }
 
+func (ts *TestSuite) CreateLabeledApplication(t *testing.T, labels map[string]string) {
+	application, err := ts.k8sClient.CreateLabeledApplication(context.Background(), ts.application, labels)
+	require.NoError(t, err)
+	require.NotNil(t, application)
+}
+
+func (ts *TestSuite) UpdateLabeledApplication(t *testing.T, labels map[string]string) {
+	application, err := ts.k8sClient.GetApplication(context.Background(), ts.application, metav1.GetOptions{})
+	require.NoError(t, err)
+	require.NotNil(t, application)
+
+	application.Spec.Labels = labels
+
+	updated, err := ts.k8sClient.UpdateApplication(context.Background(), application)
+	require.NoError(t, err)
+	require.NotNil(t, updated)
+}
+
+func (ts *TestSuite) AssertRunArgGateway(t *testing.T, expectedArg string) {
+	ts.AssertRunArg(t, ts.gateway, expectedArg)
+}
+
+func (ts *TestSuite) WaitForRunArgGateway(t *testing.T, expectedArg string) {
+	ts.WaitForRunArg(t, ts.gateway, expectedArg)
+}
+
+func (ts *TestSuite) AssertRunArg(t *testing.T, name string, expectedArg string) {
+	require.True(t, ts.containsArg(t, name, expectedArg), expectedArg)
+}
+
+func (ts *TestSuite) WaitForRunArg(t *testing.T, name string, expectedArg string) {
+	err := testkit.WaitForFunction(defaultCheckInterval, assessLabelWaitTime, func() bool {
+		return ts.containsArg(t, name, expectedArg)
+	})
+
+	require.NoError(t, err)
+}
+
+func (ts *TestSuite) containsArg(t *testing.T, name string, expectedArg string) bool {
```

Comment:
With k8sClient using previous interface for getting Deployments we can do in place casting:

```
func (ts *TestSuite) containsArg(t *testing.T, name string, expectedArg string) bool {
	deployment, err := ts.k8sClient.GetDeployment(context.Background(), name, metav1.GetOptions{})
	require.NoError(t, err)

	for _, container := range deployment.(*v1.Deployment).Spec.Template.Spec.Containers {
		for _, arg := range container.Args {
			if arg == expectedArg {
				return true
			}
		}
	}

	return false
}
```